docs/diagrams/wallet-lifecycle/wallet-state-transition.tikz

```diff
@@ -0,0 +1,138 @@
+\pgfsetlayers{background,main}%
+
+\begin{tikzpicture}[
+% STYLES FOR NODES AND ARROWS
+  every node/.style = {align=left,    % thanks to alignment we can have line breaks
+                       font=\ttfamily\small},
+        base/.style = {rectangle, draw=gray!50, rounded corners,
+                       minimum width=5cm, minimum height=1cm,
+                       text width=4.8cm, text centered, font=\sffamily\small},
+       state/.style = {base, fill=gray!10},
+         box/.style = {rectangle,
+                       minimum width=7cm,
+                       text width=6.8cm, font=\sffamily\small},
+      node distance = 25mm,
+                  > = Triangle,       % defines the arrow tip style when `->` is used
+]
+% STATES
+% We position the nodes in relation to each other.
+
+\node[state]                               (unknown)      {UNKNOWN};
+\node[state, below=of unknown]             (live)         {LIVE};
+\node[state, below=of live]                (movingFunds)  {MOVING\_FUNDS};
+\node[state, below=of movingFunds]         (closing)      {CLOSING};
+\node[state, below=of closing]             (closed)       {CLOSED};
+\node[state, left=of closed, xshift=-3cm]  (terminated)   {TERMINATED};
+
+% ARROWS
+% 1. In place when line needed to bend more than once we specified relative
+%  coordinates of the point that the arrow needs to pass.
+% 2. To label the lines we create nodes with labels which we place 5mm from the
+% beginning of the arrows (`pos=0` positions them at the beginning, `below
+% right` defines where the text should be placed in relation to that position).
+% Style of the labels is defined in `every node/.style`.
+
+\draw[->](unknown)      --  node[below right, pos=0, yshift=-5mm, align=left] {- registerNewWallet}                   (live);
+\draw[->](live)         --  node[below right, pos=0, yshift=-5mm, align=left] {- notifyWalletHeartbeatFailed\\
+                                                                               - notifyCloseableWallet\\
+                                                                               - notifyRedemptionTimeout}            (movingFunds);
+\draw[->](movingFunds)  --  node[below right, pos=0, yshift=-5mm, align=left] {- submitMovingFundsProof\\
+                                                                               - notifyMovingFundsBelowDust}         (closing);
+\draw[->](closing)      --  node[below right, pos=0, yshift=-5mm, align=left] {- notifyWalletClosingPeriodElapsed}   (closed);
+\draw[->](live)         -|  node[above right, pos=0, xshift=5mm, align=left]  {- notifyWalletHeartbeatFailed *\\
+                                                                               - notifyCloseableWallet *\\
+                                                                               - notifyRedemptionTimeout *\\
+                                                       \sffamily\footnotesize{(* only if there's no main UTXO)}}     +(10cm,0)
```

Comment:
Let's be a bit more specific:
```suggestion
                                                       \sffamily\footnotesize{(* only if there is no main UTXO for the wallet)}}     +(10cm,0)
```

Reply:
f6b035e98f64f5323ca9322a714ad85ab4e8264b